src/emu/debug/debugcmd.cpp

```diff
@@ -3909,58 +3909,80 @@ void debugger_commands::execute_memdump(const std::vector<std::string_view> &par
 
 void debugger_commands::execute_symlist(const std::vector<std::string_view> &params)
 {
-	const char *namelist[1000];
+	// Default to CPU "0" if none specified
+	const char * cpuname = (params.empty()) ? "0" : params[0].cbegin();
```

Comment:
This makes no sense at all.  Why wouldn’t you default to the CPU that’s **currently focused** in the debugger rather than the first CPU?  That’s what all the other commands do (`bpset`, `wpset`, etc.).  Always defaulting to the first CPU a very poor choice.

Also, as I’ve said elsewhere, the purpose a symbol table is being used for is not a characteristic of the symbol table, and shouldn’t be a property of the symbol table.  Requiring a model object to know all the situations in which it will be used is broken design.